### PR TITLE
update readme links

### PR DESCRIPTION
**Pull request by nawazkh**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
- Center align K8s logo and update CAPZ badges.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
+<img src="https://godoc.org/sigs.k8s.io/cluster-api-provider-azure?status.svg"></a>
+<!-- goreportcard badge -->
+<a href="https://goreportcard.com/report/kubernetes-sigs/cluster-api-provider-azure">
+<img src="https://goreportcard.com/badge/kubernetes-sigs/cluster-api-provider-azure"></a>
+<!-- join kubernetes slack channel for cluster-api-aws-provider -->
+<a href="https://kubernetes.slack.com/messages/CEX9HENG7">
+<img src="https://img.shields.io/badge/join%20slack-%23cluster--api--azure-brightgreen"></a>
+<!-- openssf badge -->
+<!-- <a href="https://bestpractices.coreinfrastructure.org/projects/5688">
+<img src="https://bestpractices.coreinfrastructure.org/projects/5688/badge"></a> -->
```
Comment (nawazkh) [inline README.md:18]: We should probably get https://www.bestpractices.dev/en added to our project?

**Review by mboersma — COMMENTED** [review]
/lgtm

Everyone loves badges, and these seem very useful. 👍🏻

File: README.md [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
 <img src="https://github.com/kubernetes/kubernetes/raw/master/logo/logo.png"  width="100">
+</p>
+<p align="center">
+<!-- go doc / reference card -->
+<a href="https://pkg.go.dev/sigs.k8s.io/cluster-api-provider-azure">
+<img src="https://godoc.org/sigs.k8s.io/cluster-api-provider-azure?status.svg"></a>
+<!-- goreportcard badge -->
+<a href="https://goreportcard.com/report/kubernetes-sigs/cluster-api-provider-azure">
+<img src="https://goreportcard.com/badge/kubernetes-sigs/cluster-api-provider-azure"></a>
+<!-- join kubernetes slack channel for cluster-api-aws-provider -->
```
Comment (CecileRobertMichon) [inline README.md:13]:
```suggestion
<!-- join kubernetes slack channel for cluster-api-azure-provider -->
```

Comment (nawazkh) [reply README.md:13]: Updated :)

**Review by mboersma — COMMENTED** [review]
/lgtm

**Review by CecileRobertMichon — COMMENTED** [review]
/approve

**Pull request merged**